search/views.py

```diff
@@ -11,7 +11,14 @@
 
 # Create your views here
 
+def all_insert(x, e, i=0):
+    return [x[0:i]+[e]+x[i:]] + all_insert(x,e,i+1) if i<len(x)+1 else []
 
+def for_each(X, e):
+    return all_insert(X[0], e) + for_each(X[1:],e) if X else []
+
+def permute(x):
+    return [x] if len(x) < 2 else for_each( permute(x[1:]) , x[0])
```

Comment:
Standard libs have already implemented this.

Reply:
@iattempt Just updated the comments.